forte/data/ontology/top.py

```diff
@@ -1179,10 +1450,76 @@ def compute_iou(self, other):
         union = self.area + other.area - intersection
         return intersection / union
 
+    def _offset_condition_check(self):
+        """
+        When the the offset is not set, this function checks the reason the
+        offset cannot be computed and raises the corresponding error.
+
+        Raises:
+            ValueError: if the grid cell is not associated with the box and
+                the box center is not set.
+            ValueError: if the grid cell is not associated with the box.
+            ValueError: if the box center is not set.
+        """
+        result_msg = (
+            "Hence, the offset of the box center from the grid cell"
+            + "center cannot be computed."
+        )
+        if not self._is_grid_associated and not self._is_box_center_set():
+            raise ValueError(
+                "The box center is not set and the grid cell center is not set."
+                + result_msg
+            )
+        elif not self._is_grid_associated:
+            raise ValueError("The grid cell center is not set." + result_msg)
+        elif not self._is_box_center_set():
+            raise ValueError("The box center is not set." + result_msg)
+
+    def _center_condition_check(self):
+        """
+        When the the center is not set, this function checks the reason the box
+        center cannot be computed and raises the corresponding error.
+
+        Raises:
+            ValueError: if the box center is not set and the grid cell center
+                is not set.
+            ValueError: if the grid cell center is not set.
+            ValueError: if the box center is not set.
+        """
+        result_msg = "Hence, the position of the box center cannot be computed."
+        if not self._is_grid_associated and not self._is_offset_set():
+            raise ValueError(
+                "The box center is not set and the grid cell center is not set."
+                + result_msg
+            )
+        elif not self._is_grid_associated:
+            raise ValueError("The grid cell center is not set." + result_msg)
+        elif not self._is_offset_set():
+            raise ValueError("The offset is not set." + result_msg)
+
+    def _is_box_center_set(self) -> bool:
+        """
+        A function checks whether the box center is set.
+
+        Returns:
+            True if the box center is set, False otherwise.
+        """
+        return self._cy is not None and self._cx is not None
+
+    def _is_offset_set(self) -> bool:
+        """
+        A function checks whether the offset of the box center from the grid
+        is set.
+
+        Returns:
+            True if the offset is set, False otherwise.
+        """
+        return self._cy_offset is not None and self._cx_offset is not None
+
 
 class BoundingBox(Box):
```

Comment:
what differences would a boundingbox have comparing to a box?

Reply:
Normally I don't think a box need be associated with a grid but it can be associated with grid anyway. A bounding box are much more often associated with a grid due to the CV algorithm.